docs/design/architecture.md

```diff
@@ -0,0 +1,29 @@
+# Architecture 
+Kubernetes Console project consists of two main components. They are called here the
+frontend and the backend.
+
+The frontend is a single page web application that runs in a browser. It fetches all its
+business data from the backend using standard HTTP methods. It does not implement business logic;
+it only presents fetched data and sends requests to the backend for actions.
+
+The backend runs in a Kubernetes cluster as a kubernetes service. Alternatively, it may run anywhere
+outside of the cluster, given that it can connect to the master. The backend is a HTTP server that
+proxies data requests to appropriate remote backends (e.g., k8s apiserver or heapster) or implements
```

Comment:
@liggitt  On the other hand, when there is no backend for the UI, the browser has to have direct access to the kubernetes master (or, at least, apiserver) and store credentials on user's computer. This essentially means that master has to be world accessible, which is not true in all setups. 

Can you describe use case for non-transitive auths? Is this something hypothetical, or solution that people actually use?